### PR TITLE
[DUOS-2046][risk=low] TDR Dataset Endpoint

### PR DESCRIPTION
**Pull request by connorlbark**
## Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2046

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/resources/TDRResource.java [hunk 1]
```diff
@@ -47,4 +47,21 @@ public Response getApprovedUsers(@Auth AuthUser authUser, @PathParam("identifier
             return createExceptionResponse(e);
         }
     }
+
+    @GET
+    @Produces("application/json")
+    @PermitAll
```
Comment (connorlbark) [inline src/main/java/org/broadinstitute/consent/http/resources/TDRResource.java:53]: is this the correct authentication scheme?

Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/resources/TDRResource.java:53]: Yes, exactly. This allows for any authenticated user to access the endpoint.

File: src/test/java/org/broadinstitute/consent/http/resources/TDRResourceTest.java [hunk 2]
```diff
@@ -56,7 +56,23 @@ public void testGetApprovedUsersForDataset() {
 
         assertEquals(200, r.getStatus());
         assertEquals(approvedUsers, r.getEntity());
+    }
+
+    @Test
+    public void testGetDatasetByIdentifier() {
```
Comment (rushtong) [inline src/test/java/org/broadinstitute/consent/http/resources/TDRResourceTest.java:73]: We could use a 404 test as well

**Review by rushtong — APPROVED** [review]
👍🏽

**Review by JVThomas — APPROVED** [review]
LGTM

**Pull request merged**